lib/_stream_duplex.js

```diff
@@ -71,6 +72,18 @@ function Duplex(options) {
 }
 
 ObjectDefineProperties(Duplex.prototype, {
+  writable: {
+    get() {
+      const w = this._writableState;
+      if (w && w.writable !== undefined) return w.writable;
+      return Boolean(w && !w.destroyed && !w.errored && !w.ending);
+    },
+    set(val) {
+      if (this._writableState) {
+        this._writableState.writable = val;
+      }
+    }
+  },
```

Comment:
Can we avoid duplicating this code between Writable and Duplex?

Reply:
Yea, I tried that... but copying getter/setters from the prototype doesn't work... this is some javascript thing I don't understand, i.e. the following doesn't work:

```js
  writable: Writable.prototype.writable
```

Is there another way?

Reply:
The getter is automatically accessed and the result is what is assigned. That's what happenes if you try to do `writable: Writable.prototype.writable`.
Instead, you need the property descriptor: `writable: Object.getOwnPropertyDescriptor(Writable.prototype, 'writable')`

Reply:
Thanks for the explanation! Fixed.